src/components/Header.js

```diff
@@ -0,0 +1,119 @@
+import React, {Component} from 'react';
+import {withStyles} from '@material-ui/core/styles';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import IconButton from '@material-ui/core/IconButton';
+import MenuIcon from '@material-ui/icons/Menu';
+import AccountCircle from '@material-ui/icons/AccountCircle';
+import MenuItem from '@material-ui/core/MenuItem';
+import Menu from '@material-ui/core/Menu';
+import Button from '@material-ui/core/Button';
+import {LogoutUser, userGetTokenFetch} from "../store/users/actions"
+import {connect} from "react-redux";
+import {Link, withRouter} from "react-router-dom";
+
+
+const styles = theme => ({
+    root: {
+        flexGrow: 1,
+    },
+    menuButton: {
+        marginRight: theme.spacing(2),
+    },
+    title: {
+        flexGrow: 1,
+    },
+});
+
+class Header extends Component {
+    state = {
+        anchorEl: null,
+        isOpened: false,
+    };
+
+    componentDidMount() {
+        if (localStorage.token === undefined) {
+            this.props.userGetTokenFetch();
+        }
+    }
+
+
+    handleOpenClick = event => {
+        this.setState(({isOpened}) => ({isOpened: !isOpened}));
+        this.setState({anchorEl: event.currentTarget});
+
+    };
+
+    render() {
+        const {classes, usersState, history, LogoutUser} = this.props;
```

Comment:
```suggestion
        const {
        classes, 
        usersState : { auth: { access } }, 
        history, 
        LogoutUser
        } = this.props;
```
in this case you can avoid adding additional constant which is on next line